### PR TITLE
DIV-6911/DIV-6533: RespSol complete AOS whilst bailiff in progress

### PR DESCRIPTION
**Pull request by ravinderpal**
**Before creating a pull request make sure that:**

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DIV-6911
related with: https://tools.hmcts.net/jira/browse/DIV-6533

### Change description ###
- added CRU permission to CAA
- added new callback

### CODE REVIEW

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json [hunk 1]
```diff
@@ -272,13 +272,6 @@
   "UserRole": "caseworker-caa",
   "CRUD": "R"
   },
-  {
-    "LiveFrom": "01/02/2021",
-    "CaseTypeID": "DIVORCE",
-    "CaseStateID": "AwaitingBailiffService",
```
Comment (ravinderpal) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json:278]: Moved to bailiff non-prod file

File: definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json [hunk 2]
```diff
@@ -75,7 +76,7 @@
     "Description": "Update Acknowledgement of Service",
     "DisplayOrder": 1,
     "PreConditionState(s)": "AosDrafted;AwaitingBailiffReferral;AwaitingBailiffService;IssuedToBailiff",
-    "PostConditionState": "AosDrafted",
+    "PostConditionState": "*",
```
Comment (ravinderpal) [inline definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json:79]: new callback not needed to handle change state here, as from each of the Precondition states it is supposed to go back to the same state

**Review by shaunganley — APPROVED** [review]

File: definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json (hunk 2 above)
Comment (Dizrupto) [reply definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json:79]: Nice

File: definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json [hunk 3]
```diff
@@ -60,9 +60,10 @@
     "Description": "Draft Acknowledgement of Service",
     "DisplayOrder": 1,
     "PreConditionState(s)": "AosAwaitingSol;AosAwaiting;AwaitingBailiffReferral;AwaitingBailiffService;IssuedToBailiff",
```
Comment (Dizrupto) [inline definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json:62]: I think to be able to run this event from the new states with a solicitor, you will need to add "CRU" permissions to the solicitor on each of those new states. Might be worth checking.

**Review by Dizrupto — APPROVED** [review]

File: definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json (hunk 3 above)
Comment (ravinderpal) [reply definitions/divorce/json/CaseEvent/CaseEvent-bailiff-nonprod.json:62]: Permissions already there 👍

File: package.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
       "ccdUrl": "http://ccd-data-store-api-demo.service.core-compute-demo.internal"
     },
     "aat": {
-      "cosUrl": "http://div-cos-aat.service.core-compute-aat.internal",
+      "cosUrl": "http://div-cos-pr-1316.service.core-compute-preview.internal",
```
Comment (ravinderpal) [inline package.json:22]: To be removed before Merging!

**Review by yarpo — APPROVED** [review]

**Comment by yarpo** [comment]
@ravinderpal why this is still opened?

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json [hunk 5]
```diff
@@ -0,0 +1,296 @@
+[
+  {
+    "LiveFrom": "01/02/2021",
+    "CaseTypeID": "DIVORCE",
+    "CaseStateID": "AosAwaiting",
```
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json:5]: please see if this (state, role) authorisation doesn't exist yet. 

It does:
```
  {
    "LiveFrom": "26/03/2021",
    "CaseTypeID": "DIVORCE",
    "CaseStateID": "AosAwaiting",
    "UserRole": "caseworker-caa",
    "CRUD": "CRU"
  },
```

in: `definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState.json:3201`

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+    "UserRole": "caseworker-caa",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/02/2021",
+    "CaseTypeID": "DIVORCE",
+    "CaseStateID": "AosStarted",
```
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json:12]: as above, and all other permissions for CAA seem to be redundant.

File: definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json (hunk 5 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseState/AuthorisationCaseState-resp-journey-nonprod.json]: yes. all this file is redundant. 
removed it.

**Review by yarpo — CHANGES_REQUESTED** [review]
I have talked to Jeremy and there will be another ticket.

**Review by yarpo — APPROVED** [review]

**Pull request merged**